module/pattern/launch/src/main/java/org/openbase/jul/pattern/launch/AbstractLauncher.java

```diff
@@ -59,21 +63,25 @@
  *
  * @author <a href="mailto:divine@openbase.org">Divine Threepwood</a>
  */
-public abstract class AbstractLauncher<L extends Launchable> extends AbstractIdentifiableController<ActivationState, ActivationState.Builder> implements Launcher, VoidInitializable, NameProvider {
-
-    protected final Logger logger = LoggerFactory.getLogger(getClass());
+public abstract class
+AbstractLauncher<L extends Launchable> extends AbstractIdentifiableController<LauncherData, LauncherData.Builder> implements Launcher, VoidInitializable, NameProvider {
 
     public static final long LAUNCHER_TIMEOUT = 60000;
     public static final String SCOPE_PREFIX_LAUNCHER = ScopeProcessor.COMPONENT_SEPARATOR + "launcher";
-
+    private static final List<Future<?>> waitingTaskList = new ArrayList<>();
+    private static final SyncObject VERIFICATION_STACK_LOCK = new SyncObject("VerificationStackLock");
+    private static final SyncObject ERROR_STACK_LOCK = new SyncObject("ErrorStackLock");
+    private static final SyncObject WAITING_TASK_LIST_LOCK = new SyncObject("WaitingStopLock");
```

Comment:
Shouldn't this SyncObject maybe be called `WaitingTaskLock` instead of `WaitingStopLock`?
```suggestion
    private static final SyncObject WAITING_TASK_LIST_LOCK = new SyncObject("WaitingTaskLock");
```